darshan-util/pydarshan/darshan/experimental/plots/data_access_by_filesystem.py

```diff
@@ -0,0 +1,591 @@
+                                                          mod=mod)
+    read_groups, write_groups = processing_func(df_reads=df_reads,
+                                                df_writes=df_writes)
+    # if either of the Series are effectively empty we want
+    # to produce a new Series with the filesystem_root indices
+    # and count values of 0 (for plotting purposes)
+    read_groups, write_groups = check_empty_series(read_groups=read_groups,
+                                                   write_groups=write_groups,
+                                                   filesystem_roots=filesystem_roots)
+    if verbose:
+        print("read_groups:\n", read_groups)
+        print("write_groups:\n", write_groups)
+    return (read_groups, write_groups)
+
+
+def plot_data(fig: Any,
+              file_rd_series,
+              file_wr_series,
+              bytes_rd_series,
+              bytes_wr_series,
+              filesystem_roots: Sequence[str]):
+    """
+    Produce the horizontal bar plots for the data
+    access by category analysis.
+
+    Parameters
+    ----------
+
+    fig: matplotlib ``figure`` object into which the
+    bar plots will be placed
+
+    file_rd_series: a ``pd.Series`` object with per-category
+    counts of unique files from which at least 1 byte has been
+    read
+
+    file_wr_series: a ``pd.Series`` object with per-category
+    counts of unique files to which at least 1 byte has been
+    written
+
+    bytes_rd_series: a ``pd.Series`` object with per-category
+    counts of bytes read
+
+    bytes_wr_series: a ``pd.Series`` object with per-category
+    counts of bytes written
+
+    filesystem_roots: a sequence of strings containing unique filesystem root paths
+    """
+    list_byte_axes: list = []
+    list_count_axes: list = []
+    # use log10 scale if range exceeds
+    # two orders of magnitude in a column
+    use_log = [False, False]
+    for idx, series_pair in enumerate([[bytes_rd_series, bytes_wr_series],
+                                       [file_rd_series, file_wr_series]]):
+        maxval = max(series_pair[0].max(), series_pair[1].max())
+        minval = min(series_pair[0].min(), series_pair[1].min())
+        ratio = maxval / minval
+        if ratio > 100:
+            use_log[idx] = True
+
+    for row, filesystem in enumerate(filesystem_roots):
+        ax_filesystem_bytes = fig.add_subplot(len(filesystem_roots),
+                                              2,
+                                              row * 2 + 1)
+        ax_filesystem_counts = fig.add_subplot(len(filesystem_roots),
+                                              2,
+                                              row * 2 + 2)
+        if row > 0:
+            ax_filesystem_bytes.get_shared_x_axes().join(ax_filesystem_bytes, list_byte_axes[row - 1])
+            if use_log[0]:
+                ax_filesystem_bytes.set_xscale('log')
```

Comment:
The code diff coverage is back to 100 % now with a second test case (well, Nik's synthetic data from another issue) that log scales both columns (other display issues notwithstanding):

![test](https://user-images.githubusercontent.com/7903078/139954611-f69902b4-1d02-4511-a0cd-f00c5c344312.png)

Reply:
The "symmetric log scaled" label addresses my original comment, and it looks like the tests are covering these cases, so I think this can be resolved.